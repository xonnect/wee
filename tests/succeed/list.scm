;; tests with lists

(define lst1 '())
(define lst2 '(1 2 3 4))
(define lst3 '(a b c d))
(define lst4 '(#\a #\b #\c #\d))

(displayln (null? lst1))
(displayln (not (null? lst2)))
(displayln (length lst1))
(displayln (length lst2))
(displayln (append lst2 lst3))
(displayln (reverse lst4))

(displayln (list->string lst4))
(define s "abcd") ; to defeat constant folding
(displayln (string->list s))

(define lst5 (list 1 2 3 4))
(list-set! lst5 2 5)
(displayln (list-ref lst5 2))
