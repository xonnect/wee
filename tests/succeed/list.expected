#t
#t
0
4
(1 2 3 4 #<symbol> #<symbol> #<symbol> #<symbol>)
(100 99 98 97)
abcd
(97 98 99 100)
5
